test/system/252-quadlet.bats

```diff
@@ -1470,4 +1470,67 @@ EOF
 
     run_podman rmi $(pause_image)
 }
+
+# This test reproduces https://github.com/containers/podman/issues/20432
+# In order to reproduce the issue, the image in the FROM must no be available locally
+# and must not have a tag. The first forces Pull and the second the resolution where the crash occurs
+# Using a local registry does not work since kube play does not pass the autofile and tls-verify flags to the build
+@test "quadlet - kube build from unavailable image with no tag" {
+    local quadlet_tmpdir=$PODMAN_TMPDIR/quadlets
+
+    local image_name=test_image
+    local yaml_dir=$quadlet_tmpdir/$image_name
+    local build_dir=$yaml_dir/$image_name
+
+    # Use the same directory for all quadlet files to make sure later steps access previous ones
+    mkdir -p $build_dir
+
+    container_file_path=$build_dir/Containerfile
+    cat >$container_file_path << EOF
+FROM busybox
+EOF
+
+    # Create the YAMl file
+    pod_name="test_pod"
+    container_name="test"
+    yaml_source="$yaml_dir/build_$(random_string).yaml"
+    cat >$yaml_source <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: test
+  name: $pod_name
+spec:
+  containers:
+  - command:
+    - "sh"
+    args:
+    - "-c"
+    - "echo STARTED CONTAINER; sleep inf"
+    image: $image_name
+    name: $container_name
+EOF
+
+    # Create the Quadlet file
+    local quadlet_file=$quadlet_tmpdir/build_$(random_string).kube
+    cat > $quadlet_file <<EOF
+[Kube]
+Yaml=${yaml_source}
+PodmanArgs=--build
+SetWorkingDirectory=yaml
+EOF
+
+    run_quadlet "$quadlet_file"
+    service_setup $QUADLET_SERVICE_NAME
+
+    # Ensure we have output.
+    wait_for_output "STARTED CONTAINER" $pod_name-$container_name
+
+    run_podman container inspect  --format "{{.State.Status}}" test_pod-test
+    is "$output" "running" "container should be started by systemd and hence be running"
+
+    service_cleanup $QUADLET_SERVICE_NAME inactive
+    run_podman rmi $(pause_image)
```

Comment:
Nasty [red](https://api.cirrus-ci.com/v1/artifact/task/6617186207268864/html/sys-podman-rawhide-rootless-host-sqlite.log.html#t--00408). Suggestion:
```diff
@@ -1480,2 +1480,3 @@ EOF
 
+    local untagged_image=quay.io/libpod/busybox
     local image_name=test_image
@@ -1489,3 +1490,3 @@ EOF
     cat >$container_file_path << EOF
-FROM busybox
+FROM $untagged_image
 EOF
@@ -1533,3 +1534,3 @@ EOF
     service_cleanup $QUADLET_SERVICE_NAME inactive
-    run_podman rmi $(pause_image)
+    run_podman rmi $untagged_image $image_name $(pause_image)
 }
```
(secondary, low-priority suggestion: rename variable `image_name` to `built_image`)

Reply:
Done (both), thanks

Reply:
Well, partly: `$untagged_image` is undefined, so there's still [red](https://api.cirrus-ci.com/v1/artifact/task/6737780382892032/html/sys-podman-rawhide-rootless-host-sqlite.log.html#t--00408). You can either take the other half of my suggestion above, or `s/$untagged_image/busybox/`. My reason for suggesting `$untagged_image` and setting it to a repo+image (untagged) is to prevent `docker.io` rate limiting on hosts like Fedora CI where they might not have our `registries.conf` overrides. I would like to prevent Fedora Gating flakes.

Obviously I tested `quay.io/libpod/busybox` on my end, using your PR as well as `main` (where the test fails with the expected crash) but if you go that route please confirm.

Reply:
I've taken in all your comments including using `quay.io/libpod/busybox`. I also make sure that the image is not available on the local machine before running the test. I ran the test locally and it did not prompt about any stray images. So, hopefully, it's OK now.

BTW initially, I wanted to use a local registry instead of downloading a new image. However, building an image based on a private registry using `kube play` is not fully supported (See #20890 for details)

Anyhow, thanks a lot for your help (here and in general)